rpcclient/wallet.go

```diff
@@ -2608,12 +2608,128 @@ func (c *Client) GetWalletInfo() (*btcjson.GetWalletInfoResult, error) {
 	return c.GetWalletInfoAsync().Receive()
 }
 
+// FutureBackupWalletResult is a future promise to deliver the result of an
+// BackupWalletAsync RPC invocation (or an applicable error)
+type FutureBackupWalletResult chan *response
+
+// Receive waits for the response promised by the future
+func (r FutureBackupWalletResult) Receive() error {
+	_, err := receiveFuture(r)
+	return err
+}
+
+// BackupWalletAsync returns an instance of a type that can be used to get the result
+// of the RPC at some future time by invoking the Receive function on the
+// returned instance.
+//
+// See BackupWallet for the blocking version and more details.
+func (c *Client) BackupWalletAsync(destination string) FutureBackupWalletResult {
+	return c.sendCmd(btcjson.NewBackupWalletCmd(destination))
+}
+
+// BackupWallet safely copies current wallet file to destination, which can
+// be a directory or a path with filename
+func (c *Client) BackupWallet(destination string) error {
+	return c.BackupWalletAsync(destination).Receive()
+}
+
+// FutureDumpWalletResult is a future promise to deliver the result of an
+// DumpWallet RPC invocation (or an applicable error)
+type FutureDumpWalletResult chan *response
+
+// Receive waits for the response promised by the future
+func (r FutureDumpWalletResult) Receive() error {
+	_, err := receiveFuture(r)
```

Comment:
The result may not contain the same `destination` as the input, so we should not ignore it.